main.tf

```diff
@@ -18,23 +35,28 @@ data "terraform_remote_state" "cluster" {
   }
 }
 
-provider "kubernetes" {
-  version = "~> 1.11"
+provider "aws" {
+  region = data.terraform_remote_state.eks_cluster.outputs.aws_region
+}
+
+data "aws_eks_cluster" "cluster" {
+  name = data.terraform_remote_state.eks_cluster.outputs.cluster
```

Comment:
Let's rename the output in the k8s workspace cluster_id so it's more descriptive?